docs/cryptodoc/src/05_09_kyber.rst

```diff
@@ -141,161 +302,115 @@ Based on these functions the key generation process follows **Algorithms 4 and 7
 
-   - ``pk = (pk_t, seed)``: public key
-   - ``m``: message
-   - ``coins``: randomness (input :math:`r` in Alg. 5 [Kyber-R3]_)
+   -  ``rng``: random number generator
+   -  ``pk = (t, rho)``: public key
 
    **Output:**
 
-   - ``c``: ciphertext bytes
+   -  ``K``: shared secret key
+   -  ``c``: ciphertext
 
    **Steps:**
 
-   1. ``at = PolynomialMatrix::generate(seed, true, mode)`` (``true`` means "transposed") (L. 3-8, Alg. 5 [Kyber-R3]_)
-   2. ``sp = PolynomialVector::getnoise_eta1(coins, 0, mode)`` (performs ``k`` invocations of ``Polynomial::getnoise_eta1``, one for each component of ``sp``; L. 9-12, Alg. 5 [Kyber-R3]_)
-   3. ``ep = PolynomialVector::getnoise_eta2(coins, k, mode)`` (performs ``k`` invocations of ``Polynomial::getnoise_eta2``, one for each component of ``ep``; L. 13-16, Alg. 5 [Kyber-R3]_)
-   4. ``epp = Polynomial::getnoise_eta2(coins, 2*k, mode)`` (L. 17, Alg. 5 [Kyber-R3]_)
-   5. ``sp.ntt()`` (L. 18, Alg. 5 [Kyber-R3]_)
-   6. ``bp = (at * sp).invntt() + ep`` (L. 19, Alg. 5 [Kyber-R3]_)
-   7. ``v = (pk_t * sp).invntt() + epp + Polynomial::from_message(m)`` (L. 20, Alg. 5 [Kyber-R3]_)
-   8. ``c = Ciphertext(bp, v, mode).to_bytes()`` (L. 21-23, Alg. 5 [Kyber-R3]_)
-
-   **Notes:**
-
-   - The member function ``Polynomial::getnoise_eta1(seed, nonce, mode)`` uses ``PRF`` on the seed with incremented nonce values to call ``Polynomial::getnoise_cbd2`` or ``Polynomial::getnoise_cbd3`` depending on ``eta1``.
-   - The member function ``Polynomial::getnoise_eta2(seed, nonce, mode)`` uses ``PRF`` on the seed with incremented nonce values to call ``Polynomial::getnoise_cbd2`` (as for all parameter sets ``eta2 = 2``).
-
-.. _pubkey/kyber/ccakem_enc:
-
-Kyber.CCAKEM
-^^^^^^^^^^^^
-
-Encapsulation works as follows, realizing **Algorithm 8** of [Kyber-R3]_:
-
-.. admonition:: Kyber_KEM_Encryptor::raw_kem_encrypt()
+   1. Generate a random message ``m`` using ``rng``
+   2. ``(K, r) = G(m || H(pk))``
+   3. K-PKE encrypt ``m`` using ``r`` to obtain ciphertext ``c``
 
-   **Input:**
-
-   - ``pk = (pk_t, seed)``: public key
-   - ``out_encapsulated_key``: ciphertext of shared key (to be overwritten)
-   - ``out_shared_key``: plaintext shared key (to be overwritten)
-   - ``rng``: random number generator
-
-   **Output:**
-
-   -  Overwritten ``out_encapsulated_key``, ``out_shared_key``
-
-   **Steps:**
+      4. Sample transposed matrix ``At`` from ``rho`` using ``sample_matrix``
+      5. Initialize a ``PolynomialSampler`` ``ps`` with ``sigma``
+      6. ``y = ntt(ps.sample_polynomial_vector_cbd_eta1())``
+      7. ``e1 = ps.sample_polynomial_vector_cbd_eta2()``
+      8. ``e2 = ps.sample_polynomial_cbd_eta2()``
+      9. ``u = inverse_ntt(At * y) + e1``
+      10. ``mu = polynomial_from_message(m)`` for byte decoding and decompression
+      11. ``v = inverse_ntt(t * y) + e2 + mu``
+      12. Encode and compress ``u`` and ``v`` to obtain ``c = c1 || c2`` using ``compress_ciphertext``
+      13. ``c = c1 || c2``
```

Comment:
```suggestion
      12. Encode, compress and concatenate ``u`` and ``v`` to obtain the
          ciphertext ``c`` using ``compress_ciphertext``
```

... no need to introduce c1 and c2 here, they aren't assigned in the pseudo code beforehand anyway.

Reply:
c